Modules/_struct.c

```diff
@@ -1734,8 +1735,8 @@ Struct_iter_unpack(PyStructObject *self, PyObject *buffer)
     }
     if (iter->buf.len % self->s_size != 0) {
         PyErr_Format(StructError,
-                     "iterative unpacking requires a bytes length "
-                     "multiple of %zd",
+                     "iterative unpacking requires a buffer of "
```

Comment:
“. . . a buffer of _a_ multiple of . . .” would read better to me. Or simpler: “requires a multiple of . . . bytes”.